### PR TITLE
refactor(web): subkey abstraction (in-browser step 2) - encapsulates subkey selection/highlighting

### PR DESCRIPTION
**Pull request by jahorton**
In 14.0, subkey highlighting and selection is managed by the same code that handles _base key_ highlighting and selection.  This is part of why https://github.com/keymanapp/keyman/pull/5288/commits/063c326f5e0e088f85287129abbc20fbb3010820 (part of #5288) removed a defunct code path.  Fortunately, that path was actually pretty functional... just disconnected. 

A better approach moving forward:  the browser subkey menu should be (mostly) responsible for managing selection of its subkeys, rather than entangling that management with core-layout code.  That entanglement would otherwise get in the way of further abstraction + modularization.

------

Arc overview:

1. #5294
    - Step 1 theme:  subkey commands as async events (as _gestures_ are inherently async)
2. #5295
3. #5354
    - Step 2 theme:  code path cleanup / clarification
4. #5355
5. #5356
    - Step 3 theme:  prep for common abstraction for the two control flows
6. #5357
    - that's right - "Gesture".  These aren't intended for _just_ longpresses, though they'll be the only supported gesture at first.
    - As gestures are events that evaluate over intervals of time, they are inherently asynchronous.  Hence, the use of `Promise`s by earlier PRs in the chain.
7. #5358
    - Given a "gesture" abstraction, establishes gesture-oriented patterns 

A rough breakdown of some of the design goals + patterns may be found in the description of #5294.

### CODE REVIEW

File: web/source/osk/browser/subkeyPopup.ts [hunk 1]
```diff
@@ -211,5 +222,30 @@ namespace com.keyman.osk.browser {
...
+
+          let onKey=(x > x0 && x < x1 && y > y0 && y < y1);
+          if(onKey) {
+            this.baseKey.key.highlight(false);
+            this.currentSelection = sk;
+          }
+          sk.key.highlight(onKey);
+        } catch(ex){}
+      }
+    }
```
Comment (jahorton) [inline web/source/osk/browser/subkeyPopup.ts:249]: This (functionally) restores the following code removed by https://github.com/keymanapp/keyman/pull/5288/commits/063c326f5e0e088f85287129abbc20fbb3010820, as referenced in this PR's description:

https://github.com/keymanapp/keyman/blob/d68fb7bac3edc50fe16c296459f61ee4dce6ac90/web/source/osk/visualKeyboard.ts#L2353-L2364

File: web/source/osk/browser/subkeyPopup.ts [hunk 2]
```diff
@@ -211,5 +222,30 @@ namespace com.keyman.osk.browser {
...
+          let x1=x0+sk.offsetWidth;
+          let y1=y0+sk.offsetHeight;
+
+          let onKey=(x > x0 && x < x1 && y > y0 && y < y1);
+          if(onKey) {
+            this.baseKey.key.highlight(false);
+            this.currentSelection = sk;
+          }
+          sk.key.highlight(onKey);
+        } catch(ex){}
```
Comment (mcdurdin) [inline web/source/osk/browser/subkeyPopup.ts:247]: I know this is copied from older code but why are we catching exceptions here?

File: web/source/osk/visualKeyboard.ts [hunk 3]
```diff
@@ -597,19 +596,18 @@ namespace com.keyman.osk {
       // Clear repeated backspace if active, preventing 'sticky' behavior.
       this.cancelDelete();
 
-      if((sk && sk.style.visibility == 'visible')) {
+      if((this.subkeyPopup && this.subkeyPopup.element.style.visibility == 'visible')) {
```
Comment (mcdurdin) [inline web/source/osk/visualKeyboard.ts:599]:
```suggestion
      if(this.subkeyPopup && this.subkeyPopup.element.style.visibility == 'visible') {
```

**Comment by jahorton** [comment]
Abandoning in favor of #5387 - a documented + combined version of all seven, as many of the changes in later steps reshuffle code from earlier steps.

**Pull request closed without merging**